python/lib/core/dmod/core/meta_data.py

```diff
@@ -467,6 +468,75 @@ def factory_init_from_deserialized_json(cls, json_obj: dict):
         except:
             return None
 
+    @classmethod
+    def factory_init_from_restriction_collections(cls, data_format: DataFormat, **kwargs) -> 'DataDomain':
+        """
+        Create and return a data domain object of the given format and keyword args containing restriction properties.
+
+        The key for each restriction determines the appropriate ::class:`StandardDatasetIndex` for the restriction's
+        ``variable`` property via ::method:`StandardDatasetIndex.get_for_name``.
+
+        The restriction property values should either be a single value, a list or a dictionary with exactly two inner
```

Comment:
This is super minor, but I think using an oxford comma here makes this a little more clear:

> [...] single value, a list, or a dictionary [...].

Reply:
Nice! Thanks for doing that. Re-reviewing / approving now.